lib/io/QuadSinkHdt.ts

```diff
@@ -0,0 +1,82 @@
+import type { WriteStream } from 'fs';
+import { createWriteStream } from 'fs';
+import * as fs from 'fs/promises';
+import * as Path from 'path';
+import * as readline from 'readline';
+import type * as RDF from '@rdfjs/types';
+import * as Docker from 'dockerode';
+import { ConcurentPromiseManager } from './ConcurentPromiseManager';
+import type { IQuadSinkFileOptions } from './QuadSinkFile';
+import { QuadSinkFile } from './QuadSinkFile';
+import { transformToHdt, pullHdtCppDockerImage } from './rfdhdtDockerUtil';
+
+export class QuadSinkHdt extends QuadSinkFile {
+  private readonly files: Set<string> = new Set();
+  private readonly deleteSourceFiles: boolean;
+  private readonly errorFileDockerRfdhdt: WriteStream;
+  private readonly hdtConcurrentTransformationManager: ConcurentPromiseManager<void>;
+
+  public constructor(
+    options: IQuadSinkFileOptions,
+    poolSize = 1,
+    deleteSourceFiles = false,
+    errorFileDockerRfdhdt = './error_log_docker_rfdhdt',
+  ) {
+    super(options);
+    this.deleteSourceFiles = deleteSourceFiles;
+    this.hdtConcurrentTransformationManager = new ConcurentPromiseManager(poolSize, []);
+    this.errorFileDockerRfdhdt = createWriteStream(errorFileDockerRfdhdt);
+  }
+
+  public async push(iri: string, quad: RDF.Quad): Promise<void> {
+    const path = Path.join('./', this.getFilePath(iri));
+    await super.push(iri, quad);
+
+    // Add files with the defined extension to the list to be transformed
+    if (this.fileExtension !== undefined && path.includes(this.fileExtension.replace('$', ''))) {
```

Comment:
```suggestion
    if (this.fileExtension && path.includes(this.fileExtension.replace('$', ''))) {
```
Please pay attention to this one. It seems to be recurring a lot.

Reply:
Why is the `.replace('$', '')` needed?

Reply:
I the config the file extension can be ` "fileExtension": "$.nq"`

Reply:
But why do we need this replacement? `$` is a valid character.

Reply:
I thought it was part of a regex replace statement and not the file extension that I would expect to be `.{something}` with nothing before the dot.

Reply:
No, that's the actual file extension. It's how CSS represents files internally on disk.